frontend/src/Patients/PatientProfileModal.jsx

```diff
@@ -0,0 +1,128 @@
+  const [isApptModalOpen, setApptModalOpen] = useState(false);
+  const [selectedAppointment, setSelectedAppointment] = useState(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      const fetchPatientInfo = async () => {
+        try {
+          const response = await api.get(`/api/user/patient/${patientId}`, { withCredentials: true });
+          setPatient(response.data);
+
+          const appointmentsResponse = await api.get(`/api/user/patient/${patientId}/appointments`, { withCredentials: true });
+          setAppointments(appointmentsResponse.data);
+        } catch (error) {
+          console.error('Error fetching patient profile:', error.response ? error.response.data : error.message);
+        }
+      };
+
+      fetchPatientInfo();
+    }
+  }, [isOpen, patientId]);
+
+
+  //Edit Profile Command
+  const handleEditPatient = (formData) => {
+    // Implement the edit functionality
+  };
+
+  const handleDeletePatient = async () => {
+    try {
+      await api.delete(`/api/user/patient/${patientId}`, { withCredentials: true });
+      onClose();
+    } catch (error) {
+      console.error('Error deleting patient:', error.response ? error.response.data : error.message);
+    }
+  };
+
+  const handleSubmitAppointment = async (e, appointmentData) => {
+    e.preventDefault();
+    try {
+      if (selectedAppointment) {
+        const response = await api.put(`/api/user/appointments/${selectedAppointment.id}`, appointmentData, { withCredentials: true });
+        setAppointments(appointments.map(appt => appt.id === response.data.id ? response.data : appt));
+        setSelectedAppointment(null);
+        alert('Appointment Updated Successfully');
+      } else {
+        const response = await api.post('/api/user/appointments', { ...appointmentData, patientId: patient.id }, { withCredentials: true });
+        setAppointments([...appointments, response.data]);
+        alert('Appointment Created Successfully');
+      }
+      setApptModalOpen(false);
+    } catch (error) {
+      console.error('Error creating appointment:', error.response ? error.response.data : error.message);
+      alert('An error occurred while creating the appointment. Please try again.');
+    }
+  };
+
+  const handleEditAppointment = (appointment) => {
+    setSelectedAppointment(appointment);
+    setApptModalOpen(true);
+  };
+
+  const handleDeleteAppointment = async (id) => {
+    try {
+      await api.delete(`/api/user/appointments/${id}`, { withCredentials: true });
+      setAppointments(appointments.filter(appointment => appointment.id !== id));
+    } catch (error) {
+      console.error('Error deleting appointment:', error.response ? error.response.data : error.message);
+    }
+  };
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <button className="close-button" onClick={onClose}>Close</button>
+        {patient && (
```

Comment:
Is there a null state design for any text if the physician has no patients?

Reply:
Not yet, I'm yet to implement that.